test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Csv.Tests.ps1

```diff
@@ -45,27 +45,44 @@ Describe "ConvertTo-Csv" -Tags "CI" {
     }
 
     It "Should output an array of objects" {
-        $result = $testObject | ConvertTo-Csv
+        $result = $testObject | ConvertTo-Csv -IncludeTypeInformation
         ,$result | Should BeOfType "System.Array"
     }
 
     It "Should return the type of data in the first element of the output array" {
-	$result = $testObject | ConvertTo-Csv
+	$result = $testObject | ConvertTo-Csv -IncludeTypeInformation
 
 	$result[0] | Should Be "#TYPE System.Management.Automation.PSCustomObject"
     }
 
     It "Should return the column info in the second element of the output array" {
-	$result = $testObject | ConvertTo-Csv
+	$result = $testObject | ConvertTo-Csv -IncludeTypeInformation
 
 	$result[1] | Should Match "`"FirstColumn`""
 	$result[1] | Should Match "`"SecondColumn`""
     }
 
     It "Should return the data as a comma-separated list in the third element of the output array" {
-	$result = $testObject | ConvertTo-Csv
+	$result = $testObject | ConvertTo-Csv -IncludeTypeInformation
 	$result[2] | Should Match "`"Hello`""
 	$result[2] | Should Match "`"World`""
     }
 
+    It "Includes type information when -IncludeTypeInformation is supplied" {
+        $result = $testObject | ConvertTo-Csv -IncludeTypeInformation
+
+        ($result -split ([Environment]::NewLine))[0] | Should BeExactly "#TYPE System.Management.Automation.PSCustomObject"
```

Comment:
I think we need add reverse test to check that ""#TYPE " is absent if `-IncludeTypeInformation` is absent.
Also that `-NoTypeInformation` do nothing.

Reply:
sure I'll add those and do the same for the `Export-Csv` tests

Reply:
fixed